### PR TITLE
add check plugin is not used before rm

### PR DESCRIPTION
**Pull request by vieux**
ping @anusha-ragunathan @tiborvass

### CODE REVIEW

File: volume/store/store.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ import (
 	"github.com/Sirupsen/logrus"
 	"github.com/boltdb/bolt"
 	"github.com/docker/docker/pkg/locker"
+	pstore "github.com/docker/docker/plugin/store"
```
Comment (anusha-ragunathan) [inline volume/store/store.go]: nit: pstore -> pluginstore.

**Comment by anusha-ragunathan** [comment]
LGTM

**Comment by tiborvass** [comment]
I don't think I like that the client of `volumedrivers` has to do extra work because of this refcounting. Can't we somehow just say `volumedrivers.Remove(v)` etc. and have volumedrivers call a private function with the appropriate `mode` setting?

**Comment by anusha-ragunathan** [comment]
@vieux : rebase needed.

**Comment by vieux** [comment]
@anusha-ragunathan PTAL

**Comment by anusha-ragunathan** [comment]
LGTM

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**